core/ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/core/WsAgentURLProviderImpl.java

```diff
@@ -0,0 +1,45 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2016 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.ide.core;
+
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+
+import org.eclipse.che.api.machine.gwt.client.WsAgentUrlProvider;
+import org.eclipse.che.ide.api.app.AppContext;
+import org.eclipse.che.ide.util.loging.Log;
+
+/**
+ * @author Dmitry Shnurenko
+ */
+@Singleton
+public class WsAgentURLProviderImpl implements WsAgentUrlProvider {
```

Comment:
We have some modules in which we can't add dependency to use AppContext. For example platform-api-client-gwt.